tests/test_core_utils.py

```diff
@@ -24,3 +25,21 @@ def test_accessing_hydra_config(hydra_restore_singletons: Any) -> Any:
         del cfg["hydra"]
     assert cfg.job_name == "UNKNOWN_NAME"
     assert cfg.config_name == "accessing_hydra_config"
+
+
+def test_py_version_resolver(hydra_restore_singletons: Any) -> Any:
+    utils.setup_globals()
+    vi = sys.version_info
```

Comment:
it would be cleaner to mock the version.
this is similar to  what I have done at https://github.com/antlr/antlr4/pull/2866/files#diff-5895bc8c44907952f2f9f6e4e212df6cR12.

You can probably use Pytest's monkeypatch to do this more cleanly.

Reply:
looks like monkeypatch is not actually working here? (see CI failures)

Reply:
yea. It passed locally on my pytest somehow. Im fixing this :) (I've turned the PR into a draft)